DPG/Tasks/FT0/qaFT0TrgBased.cxx

```diff
@@ -0,0 +1,650 @@
+      auto ft0 = collision.foundFT0();
+
+      std::bitset<8> triggers = ft0.triggerMask();
+      OrAFT0 = triggers[o2::ft0::Triggers::bitA];
+      OrCFT0 = triggers[o2::ft0::Triggers::bitC];
+      OrACFT0 = (OrAFT0 && OrCFT0);
+      TVX = triggers[o2::ft0::Triggers::bitVertex];
+      Cent = triggers[o2::ft0::Triggers::bitCen];
+      SemiCent = triggers[o2::ft0::Triggers::bitSCen];
+      MinBias = (TVX && (Cent || SemiCent));
+
+      if (OrAFT0) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kOrAFT0);
+      }
+      if (OrCFT0) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kOrCFT0);
+      }
+      if (OrACFT0) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kOrACFT0);
+      }
+      if (TVX) {
+        histos.fill(HIST("Colls/EventSelectionStats"), k0TVX);
+      }
+      if (Cent) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kCent);
+      }
+      if (SemiCent) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kSemiCent);
+      }
+      if (MinBias) {
+        histos.fill(HIST("Colls/EventSelectionStats"), kMinBias);
+      }
+
+      // Macro for filling histograms for 'observable' based on conditions for
+      // CollisionsTask
+#define FillConditionHistos(observable, ...)                           \
+  if (hasFT0) {                                                        \
+    histos.fill(HIST("Colls/" observable "/HasFT0"), __VA_ARGS__);     \
+    if (OrAFT0) {                                                      \
+      histos.fill(HIST("Colls/" observable "/FT0OrA"), __VA_ARGS__);   \
+    }                                                                  \
+    if (OrCFT0) {                                                      \
+      histos.fill(HIST("Colls/" observable "/FT0OrC"), __VA_ARGS__);   \
+    }                                                                  \
+    if (OrACFT0) {                                                     \
+      histos.fill(HIST("Colls/" observable "/FT0OrAC"), __VA_ARGS__);  \
+    }                                                                  \
+    if (TVX) {                                                         \
+      histos.fill(HIST("Colls/" observable "/0TVX"), __VA_ARGS__);     \
+    }                                                                  \
+    if (Cent) {                                                        \
+      histos.fill(HIST("Colls/" observable "/Cent"), __VA_ARGS__);     \
+    }                                                                  \
+    if (SemiCent) {                                                    \
+      histos.fill(HIST("Colls/" observable "/SemiCent"), __VA_ARGS__); \
+    }                                                                  \
+    if (MinBias) {                                                     \
+      histos.fill(HIST("Colls/" observable "/MinBias"), __VA_ARGS__);  \
+    }                                                                  \
+  }
```

Comment:
there will be too many similar `if`s in macro callings, i.e. you can use if block for `OrAFT0` only once and fill there all necessary hists. Please use it carefully in prod (e.g. Hyperloop, etc)